mesonpy/__init__.py

```diff
@@ -601,6 +601,10 @@ def __init__(
             self._meson_args[key] = args_from_config + tuple(self._meson_args[key])
         # XXX: We should validate the user args to make sure they don't conflict with ours.
 
+        self._check_for_unknown_config_keys({
+            'args': typing_get_args(MesonArgsKeys),
+        })
```

Comment:
I'm not sure I understand the signature of this method. Can the passed dict have any other key than `args`? Where does `args` come from?

Reply:
Yep, it could be something other than `args`, if we add more config settings in the future.

Eg.

```python
self._check_for_unknown_config_keys({
    'args': typing_get_args(MesonArgsKeys),
    'other-section': ('setting1', 'setting2'),
})
```

The args key, validates the args table of the config

```
[tool.meson-python.args]
dist = []
setup = []
compile = []
install = []
invalid = []
```

The first 4 keys will be okay, but an error will be generated for `invalid`.

Similarly, according to the snippet above, the following config will be valid

```
[tool.meson-python.other-section]
setting1 = []
setting2 = []
```

Reply:
I see. I was confused by this thing taking arguments at all. See below.